docs/containerization.md

```diff
@@ -1,61 +1,97 @@
+as simple as `make cuda`.
+
+The default image name is `localhost/instructlab`, but you can override it
+with the `CONTAINER_PREFIX` environment variable. The CUDA version can be change
+with `CUDA_VERSION`.
+
+### Configure Podman with NVIDIA container runtime
+
+To configure your machine, you can follow NVIDIA's documentation to
+[install NVIDIA container toolkit](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html#installing-with-yum-or-dnf)
+and to [configure Container Device Interface (CDI)](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/cdi-support.html)
+to expose the GPUs to Podman.
+
+Here is a quick procedure if you haven't.
+
+```shell
+curl -s -L https://nvidia.github.io/libnvidia-container/stable/rpm/nvidia-container-toolkit.repo | sudo tee /etc/yum.repos.d/nvidia-container-toolkit.repo
+sudo dnf config-manager --enable nvidia-container-toolkit-experimental
+sudo dnf install -y nvidia-container-toolkit
 ```
 
-Or image: TBD (am I allowed to have a public image with references to lab in it?)
+Then, you can verify that NVIDIA container toolkit can see your GPUs.
+
+```shell
+nvidia-ctk cdi list
+```
 
-This `Containerfile` is based on Nvidia CUDA image, which lucky for us plugs
-directly into Podman via their `nvidia-container-toolkit`! The `ubi9` base
-image does not have most packages installed. The bulk of the `Containerfile` is
-spent configuring your system so `ilab` can be installed and run properly.
-`ubi9` as compared to `ubuntu` cannot install the entire `nvidia-12-4` toolkit.
-This did not impact performance during testing.
+Example output:
 
 ```shell
-1. podman build -f <Containerfile_Path>
-2. curl -s -L https://nvidia.github.io/libnvidia-container/stable/rpm/nvidia-container-toolkit.repo |   sudo tee /etc/yum.repos.d/nvidia-container-toolkit.repo
-3. sudo yum-config-manager --enable nvidia-container-toolkit-experimental
-4. sudo dnf install -y nvidia-container-toolkit
-5. sudo nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml
-6. nvidia-ctk cdi list
-    Example output: 
-    INFO[0000] Found 2 CDI devices
-    nvidia.com/gpu=0
-    nvidia.com/gpu=all
-7. podman run --device nvidia.com/gpu=0  --security-opt=label=disable -it <IMAGE_ID>
+INFO[0000] Found 2 CDI devices
+nvidia.com/gpu=0
+nvidia.com/gpu=all
 ```
 
-Voila! You now have a container with CUDA and GPUs enabled!
+Finally, you can generate the CDI configuration for the NVIDIA devices:
 
-### Sources
+```shell
+sudo nvidia-ctk cdi generate --output=/etc/cdi/nvidia.yaml
+```
+
+### Run the CUDA-accelerated InstructLab container
+
+When running our model, we want to create some paths that will be mounted in
+the container to provide data persistence. As an unprivileged user, you will
+run a rootless container and map your UID to UID `1001` inside the container
+with `--userns keep-id:uid=1001`. The volume mount will contain InstructLab's
```

Comment:
IMO it would be better to drop the USER= assign and just let this default to the container root user, then we don't need users to manually specify a userns.

Reply:
The container is now using `USER 0` again. We can switch the container to rootless at a later point in time.